### ISSUE TITLE
[dev.icinga.com #8458] Use aria-label for link purpose

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1392]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/8458

**Created by elippmann on 2015-02-17 08:50:35 +00:00**

Assignee: _jmeyer_
Status: _Resolved_  (closed on _2015-02-24 10:37:06 +00:00_)
Target Version: _2.0.0-beta3_
Last Update: _2015-02-24 10:37:06 +00:00 (in Redmine)_

---

http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140916/ARIA8 wrote:
> The objective of this technique is to describe the purpose of a link using the aria-label attribute. The aria-label attribute provides a way to place a descriptive text label on an object, such as a link, when there are no elements visible on the page that describe the object. If descriptive elements are visible on the page, the aria-labelledby attribute should be used instead of aria-label. Providing a descriptive text label lets a user distinguish the link from links in the Web page that lead to other destinations and helps the user determine whether to follow the link. In some assistive technologies the aria-label value will show in the list of links instead of the actual link text. \[...\]

**Changesets**

_2015-02-17 13:02:35 +00:00_ by _jmeyer_ a9d767867d577bc1a61811b2bba9e11e15fe16c5

    WCAG/2.4.4: Add example for link contexts
    
    refs #8458

_2015-02-23 15:50:31 +00:00_ by _jmeyer_ 64a4dc67bc758359855750335676a4005c2bb762

    Add proper titles to all config tabs
    
    refs #8458

_2015-02-23 15:56:49 +00:00_ by _jmeyer_ f6cbc17ff7c35d7ffb5f36ed709717830c6e207d

    Add proper titles to all host/service tabs
    
    refs #8458

_2015-02-23 15:57:06 +00:00_ by _jmeyer_ ebf8d183a7f44948bccf7774649486ae87480c5e

    Add proper title to the alert summary's tab
    
    refs #8458

_2015-02-23 15:57:32 +00:00_ by _jmeyer_ 77b60588bc12f66b24df1e550946a474cc134d69

    Add proper titles to the multi selection controller tabs
    
    refs #8458

_2015-02-23 15:58:48 +00:00_ by _jmeyer_ 7aacfcf907aaa4e7b411ef1e3f26cd2b72388a9e

    Add proper title to the tactical overview's tab
    
    refs #8458

_2015-02-23 15:59:13 +00:00_ by _jmeyer_ 1ef03a76f751019d06f740501247cacde09ac380

    Add proper title to the monitoring health's tab
    
    refs #8458

_2015-02-23 15:59:47 +00:00_ by _jmeyer_ f48dbf16405e330560bb878b813060af12c0ad0c

    Add proper title to the timeline's tab
    
    refs #8458

_2015-02-23 16:00:30 +00:00_ by _jmeyer_ 6eb638f6b32ee37adde3ffb5e4164e44d701ccb8

    Add proper titles to the list view's tabs
    
    refs #8458

_2015-02-23 16:04:40 +00:00_ by _jmeyer_ c6b61617eec4702b1bebf96dfa8095082e278255

    Add proper titles to the monitoring module's config tabs
    
    refs #8458

_2015-02-23 16:05:02 +00:00_ by _jmeyer_ 66305bd79a188cec81375db50d7d84920d10e787

    Add proper titles to dashboards and dashlets
    
    refs #8458

_2015-02-23 16:05:39 +00:00_ by _jmeyer_ b2f817925b760b8d9ae9033c11fb7c5165b810d8

    Add proper titles to the FilterEditor
    
    refs #8458

_2015-02-23 16:06:01 +00:00_ by _jmeyer_ fb7640db0a7b6fcb5cc60f9e3f2c567c4da9eca9

    Add proper titles to the Limiter control
    
    refs #8458

_2015-02-23 16:07:21 +00:00_ by _jmeyer_ bd9d06574d83f7a519b43681237b42b6eaddd725

    Ensure that the link() view helper adds proper titles
    
    refs #8458

_2015-02-23 16:09:37 +00:00_ by _jmeyer_ 5fd447394c2d7000dc3511ca8a6ed94660cb3e2f

    Add proper title to the BackendConfigForms's jump link...
    
    ...and move it to the right of the respective form control
    
    refs #8458

_2015-02-23 16:10:52 +00:00_ by _jmeyer_ 77fa2f7c2b5634c0e72ea89b847448a1294d6014

    Add aria-label to the monitoring health's link to disable notifications
    
    refs #8458

_2015-02-23 16:12:34 +00:00_ by _jmeyer_ a1e7cd8906d4186132d68a238cb734a009731ce7

    Add proper titles to the documentation tocs and search results
    
    refs #8458

_2015-02-23 16:14:40 +00:00_ by _jmeyer_ b626a96e162308e367040aa137b7126450ec19d3

    Add aria-label to each box of the event grid
    
    refs #8458

_2015-02-23 16:17:48 +00:00_ by _jmeyer_ de744fdc83f435e6f72d26ece2af4b1747ebe7b2

    Add proper title to each module documentation link
    
    refs #8458

_2015-02-23 16:18:32 +00:00_ by _jmeyer_ fb28da8f239deabbb9404e969f23545b907d8bf4

    Add proper titles to the documentation module's overview
    
    refs #8458

_2015-02-23 16:19:06 +00:00_ by _jmeyer_ 3605880fd1ba3dbd99ee6cb0ad728688caf596c1

    Add proper titles to the module overview
    
    refs #8458

_2015-02-23 16:19:37 +00:00_ by _jmeyer_ baf47fef7f7acb869e2e2169726ce76cf14e5411

    Add proper titles for the links to dis-/enable modules
    
    refs #8458

_2015-02-23 16:20:26 +00:00_ by _jmeyer_ 508322cfdc790bfdab185a1fe1715673a7687f68

    Add proper titles to the dashboard configuration's links
    
    refs #8458

_2015-02-23 16:20:48 +00:00_ by _jmeyer_ 261abbd65e4c974a3f7e122baab570c961df4984

    Add proper titles to the role configuration
    
    refs #8458

_2015-02-23 16:21:12 +00:00_ by _jmeyer_ a966f543f6fe755197ea3fa45c266b4e2494e844

    Add proper titles to the joystick navigation
    
    refs #8458

_2015-02-23 16:21:31 +00:00_ by _jmeyer_ 371f8951889467e8bbb40b10ff7d5b4ff5774ea2

    Add proper titles to the mixed pagination
    
    refs #8458

_2015-02-23 16:22:07 +00:00_ by _jmeyer_ 96c7b22c16c0319abf9144482356f5ac73d61df9

    Add proper titles to the host multi selection view
    
    refs #8458

_2015-02-23 16:22:52 +00:00_ by _jmeyer_ abcac2917cfc158d23a8d8c81e03b114cf5a08cd

    Add proper titles to the hostsummary component view
    
    refs #8458

_2015-02-23 16:23:09 +00:00_ by _jmeyer_ 1bcac557783ac2cad380e41ed58e604bb277a55d

    Add proper titles to the servicesummary component view
    
    refs #8458

_2015-02-23 16:23:26 +00:00_ by _jmeyer_ 3aa1f7e58eed1dd5e3f9a8ebb81648bdb45ef7e9

    Add proper titles to the contactgroups view
    
    refs #8458

_2015-02-23 16:23:39 +00:00_ by _jmeyer_ e0164e8bcb95c42a079c8cc4a4021a11fa5f551c

    Add proper titles to the contacts view
    
    refs #8458

_2015-02-23 16:23:57 +00:00_ by _jmeyer_ 45e1406e2a71a1b58854d7bd111533693fe35c69

    Add proper titles to the hostgroups view
    
    refs #8458

_2015-02-23 16:24:22 +00:00_ by _jmeyer_ a64f546a7132d845d6dd8d0bd348770c579fbd08

    Add proper titles to the host list view
    
    refs #8458

_2015-02-23 16:24:46 +00:00_ by _jmeyer_ c4aa02c3f00658a2521ad6cc25871723086fc5a9

    Add proper titles to the service grid
    
    refs #8458

_2015-02-23 16:25:02 +00:00_ by _jmeyer_ ebdf68550b86e53d21268226c2077ec7daac1c8f

    Add proper titles to the servicegroups view
    
    refs #8458

_2015-02-23 16:25:23 +00:00_ by _jmeyer_ f11ad1183a7260a477bd3135153dd9bac9e97658

    Add proper titles to the service list view
    
    refs #8458

_2015-02-23 16:25:55 +00:00_ by _jmeyer_ 598294d2a2041e22cdaaedac7e8069b240127be2

    Add proper titles to the servicesummary partial
    
    refs #8458

_2015-02-23 16:26:21 +00:00_ by _jmeyer_ 74360ee8e616a7dbcb4a8b3877294396205e5a39

    Add proper titles to the acknowledgements component view
    
    refs #8458

_2015-02-23 16:26:42 +00:00_ by _jmeyer_ 25e2baa566d413e46d2d3aee6fad64d1af631f1e

    Add proper titles to the checkstatistics component view
    
    refs #8458

_2015-02-23 16:26:56 +00:00_ by _jmeyer_ 89ed498069e8b53bcdf4a23979d41b49d58384e4

    Add proper titles to the command component view
    
    refs #8458

_2015-02-23 16:27:13 +00:00_ by _jmeyer_ 9793bd52c7033995d635eefb2617330bcef12348

    Add proper titles to the comments component view
    
    refs #8458

_2015-02-23 16:27:27 +00:00_ by _jmeyer_ 3a74ec3d09ed746039baf6c4e912d56c16bad017

    Add proper titles to the contacts component view
    
    refs #8458

_2015-02-23 16:27:45 +00:00_ by _jmeyer_ 0d8bf462f5142fbc69ec59ef13aea99adc8bc44b

    Add proper titles to the downtime component view
    
    refs #8458

_2015-02-23 16:28:02 +00:00_ by _jmeyer_ 7247ea5191ad34231468c7726918a5588c9697e9

    Add proper titles to the hostgroups component view
    
    refs #8458

_2015-02-23 16:28:20 +00:00_ by _jmeyer_ 18f6ff8702755ac0eaad9de5bf0dd8b3a08aaf41

    Add proper titles to the servicegroups component view
    
    refs #8458

_2015-02-23 16:29:22 +00:00_ by _jmeyer_ 8991b7ed83d779cff61d890d2e9d887161d1986c

    Add proper titles to the tactical overview
    
    refs #8458

_2015-02-23 16:29:50 +00:00_ by _jmeyer_ 32e048693aa42a4b055b2cee798cfb34b2e047d8

    Add proper titles to service multi selection view
    
    refs #8458

_2015-02-23 16:30:37 +00:00_ by _jmeyer_ bb5799434c909235fc26ec211f78e543e815f723

    Add proper titles to the contact view
    
    refs #8458

_2015-02-23 16:30:51 +00:00_ by _jmeyer_ 52c17805c9415cd2401e2305b6acc3fa73c023e4

    Add proper titles to the history view
    
    refs #8458

_2015-02-23 16:31:24 +00:00_ by _jmeyer_ be293f5e49e55015baeabacbdce1238262a872ea

    Add proper titles to the timeline
    
    refs #8458

_2015-02-23 16:32:55 +00:00_ by _jmeyer_ 60a9d1f224b9a5ae46a4be7df4c13452206ab340

    Wizard: Add proper title to the refresh button of the requirements page
    
    refs #8458

_2015-02-23 16:33:23 +00:00_ by _jmeyer_ e6957967f9afacadedab033d401ad5e4424e9528

    Wizard: Add proper titles to the button-like links on the last page
    
    refs #8458

_2015-02-24 08:03:47 +00:00_ by _jmeyer_ 907134cee2fb05fe02843b1941c172ab0b52ff1a

    Ensure that all link-like form buttons got proper titles
    
    refs #8458

_2015-02-24 10:35:56 +00:00_ by _jmeyer_ decab543b0c71348e9de34de656579fe141cdfd2

    Merge branch 'feature/link-labels-8458'
    
    resolves #8458
